src/test/ensemble_flow_custom_node_tests.cpp

```diff
@@ -188,6 +189,197 @@ TEST_F(EnsembleFlowCustomNodePipelineExecutionTest, AddSubCustomNode) {
     });
 }
 
+class EnsembleFlowCustomNodeAndDemultiplexerGatherPipelineExecutionTest : public EnsembleFlowCustomNodePipelineExecutionTest {
+protected:
+    ConstructorEnabledModelManager modelManager;
+    ModelConfig config = DUMMY_MODEL_CONFIG;
+    CustomNodeLibraryManager manager;
+    NodeLibrary differentOpsLibrary;
+    NodeLibrary chooseMaxLibrary;
+    const std::string differentOpsLibraryName{"different_ops"};
+    const std::string chooseMaxLibraryName{"choose_max"};
+    const std::string differentOpsLibraryPath{"/ovms/bazel-bin/src/lib_node_perform_different_operations.so"};
+    const std::string chooseMaxLibraryPath{"/ovms/bazel-bin/src/lib_node_choose_maximum.so"};
+    const std::string pipelineInputName = "pipeline_input";
+    const std::string pipelineOutputName = "pipeline_output";
+    const std::string pipelineFactorsName = "pipeline_factors";
+    const std::string chooseMaxInputName = "input_tensors";
+    const std::string chooseMaxOutputName = "maximum_tensor";
+    const std::string differentOpsInputName = "input_numbers";
+    const std::string differentOpsFactorsInputName = "op_factors";
+    const std::string differentOpsOutputName = "different_ops_results";
+    const std::string differentOpsFactorsOutputName = "different_ops_factors_results";
+    const std::unordered_map<std::string, std::string> differentOpsOutputAlias{{differentOpsOutputName, differentOpsOutputName}};
+    const std::unordered_map<std::string, std::string> chooseMaxOutputAlias{{chooseMaxOutputName, chooseMaxOutputName}};
+    const std::string dummyNodeName = "dummy";
+    const std::string differentOpsNodeName{"different-ops-node"};
+    const std::string chooseMaxNodeName{"choose-max-node"};
+    const uint32_t demultiplyCount = 4;  // different ops library has (1,4,10) as output
+
+    void SetUp() override {
+        config.setNireq(16);
```

Comment:
Why do we need nireq=16?

Reply:
we just need it higher than default 1 - will add comment in test